From 4b16ab74d20fde07c187861816eb7e65f3e76cae Mon Sep 17 00:00:00 2001
From: Jordan Williams <jordan@jwillikers.com>
Date: Fri, 31 May 2024 08:43:16 -0500
Subject: [PATCH] Require directx-headers when building wsi_common_win32.cpp

The src/vulkan/wsi/wsi_common_win32.cpp file includes directx/d3d12.h, but does not always link directx-headers.
This is problematic when attempting to build the swrast Vulkan driver without one of with_gallium_d3d12, with_microsoft_cl, with_microsoft_vk.
It may also be problematic when attempting to build the zink Gallium driver without one of those.
This commit fixes the checks for the directx-headers dependency to be required in these additional situations where wsi_common_win32.cpp is compiled.
---
 meson.build | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/meson.build b/meson.build
index 6b22930eb5f..8df481ea4d9 100644
--- a/meson.build
+++ b/meson.build
@@ -582,13 +582,13 @@ if with_gallium_kmsro and (with_platform_x11 and not with_dri3)
 endif
 
 dep_dxheaders = null_dep
-if with_gallium_d3d12 or with_microsoft_clc or with_microsoft_vk
+if with_gallium_d3d12 or with_microsoft_clc or with_microsoft_vk or (with_platform_windows and (with_any_vk or with_gallium_zink))
   dep_dxheaders = dependency('directx-headers', required : false)
   if not dep_dxheaders.found()
     dep_dxheaders = dependency('DirectX-Headers',
       version : '>= 1.613.0',
       fallback : ['DirectX-Headers', 'dep_dxheaders'],
-      required : with_gallium_d3d12 or with_microsoft_vk
+      required : with_gallium_d3d12 or with_microsoft_vk or (with_platform_windows and (with_any_vk or with_gallium_zink))
     )
   endif
 endif
-- 
2.45.1

